src/command_modules/azure-cli-cognitiveservices/azure/cli/command_modules/cognitiveservices/_params.py

```diff
@@ -69,13 +73,58 @@ def _is_suspected_json(string):
 )
 
 
+def _sku_filter(cmd, namespace):
+    """
+    Get a list of ResourceSku and filter by existing conditions: 'kind', 'location' and 'sku_name'
+    """
+    kind = getattr(namespace, 'kind', None)
+    location = getattr(namespace, 'location', None)
+    sku_name = getattr(namespace, 'sku_name', None)
+
+    def _filter_sku(_sku):
+        if sku_name is not None:
+            if _sku.name != sku_name:
+                return False
+        if kind is not None:
+            if _sku.kind != kind:
+                return False
+        if location is not None:
+            if location.lower() not in [x.lower() for x in _sku.locations]:
+                return False
+        return True
+
+    return [x for x in cf_resource_skus(cmd.cli_ctx).list() if _filter_sku(x)]
+
+
+@Completer
+def sku_name_completer(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    names = set([x.name for x in _sku_filter(cmd, namespace)])
+    return sorted(list(names))
+
+
+@Completer
+def kind_completer(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    kinds = set([x.kind for x in _sku_filter(cmd, namespace)])
+    return sorted(list(kinds))
+
+
+@Completer
+def location_completer(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    names = set(sum([x.locations for x in _sku_filter(cmd, namespace)], []))
```

Comment:
`set([location for x in _sku_filter(cmd, namespace) for location in x.locations])` to be efficient

Reply:
ahh, true!